src/OrthoTree/morton_codes.h

```diff
@@ -429,34 +435,30 @@ morton_code Morton<Dim>::get_deepest_last_descendant(morton_code code) const
     return get_last_descendant(code, max_depth);
 }
 
+
+// suggestion for alternative algorithm:
+// only go up the levels on the larger node and check whether the other is descendant.
 template <size_t Dim>
 morton_code Morton<Dim>::get_nearest_common_ancestor(morton_code code_a, morton_code code_b) const
 {
-    morton_code ancestor_a = code_a;
-    morton_code ancestor_b = code_b;
 
-    size_t depth_a = get_depth(ancestor_a);
-    size_t depth_b = get_depth(ancestor_b);
+    size_t depth_a = get_depth(code_a);
+    size_t depth_b = get_depth(code_b);
 
-    // bring both nodes to the same depth
-    while ( depth_a > depth_b ) {
-        ancestor_a = get_parent(ancestor_a);
-        depth_a--;
+    // swap nodes such that b is the coarser nodes
+    if ( depth_a < depth_b ) {
+        std::swap(code_a, code_b);
+        std::swap(depth_a, depth_b);
     }
 
-    while ( depth_b > depth_a ) {
-        ancestor_b = get_parent(ancestor_b);
-        depth_b--;
-    }
 
+    morton_code ancestor_b = code_b;
     // climp up until common ancestor is found
-    while ( ancestor_a != ancestor_b ) {
-        assert(ancestor_a != 0 && ancestor_b != 0 && "cant ascend more than root node");
-        ancestor_a = get_parent(ancestor_a);
+    while ( !is_descendant(code_a, ancestor_b) ) {
```

Comment:
maybe shorten while loop for depth(code_a) < depth(code_b) by:
code_a = get_ancestor_at_level(code_a, depth(b)-depth(a))

Reply:
Yes to avoid this I just swap the codes such that code_b is always the coarser